### PR TITLE
Plans: Nux checkout auto-follow for all

### PR DESCRIPTION
**Pull request by artpi**
Ends the test introduced in https://github.com/Automattic/wp-calypso/pull/7589 and enables the behavior for all users

More about the test in testing post:
p3fqKv-3rT-p2

## Testing

- Use incognito (to be a new user)
- Go through checkout, selecting a free plan
- See that behaviour is not changed
- Go through checkout, selecting any plan
- See that button "clicks itself"
- using your existing user name, log in
- create new site **with paid domain and a free plan**
- see that button clicks itself

CC @lamosty @meremagee @rralian @gwwar 

Test live: https://calypso.live/?branch=update/nux-auto-redirect

### CODE REVIEW

File: client/signup/main.jsx [hunk 1]
```diff
@@ -106,7 +105,7 @@ const Signup = React.createClass( {
 					: undefined;
 				const filteredDestination = utils.getDestination( destination, dependencies, this.props.flowName );
 				let loadingScreenDelay;
-				if ( dependencies.cartItem && abtest( 'signupCheckoutRedirect' ) === 'auto' ) {
+				if ( dependencies.cartItem ) {
 					loadingScreenDelay = MINIMUM_TIME_LOADING_SCREEN_IS_DISPLAYED_PREMIUM;
```
Comment (rralian) [inline client/signup/main.jsx]: Can we refactor this bit? You had added two constants for a minimum loading time so we could target the "paid user" case and leave the "free user" case as-is. But I think we may as well just go back to one constant and change it to three seconds.

Comment (artpi) [inline client/signup/main.jsx]: Sure!

Comment (artpi) [inline client/signup/main.jsx]: Done

**Comment by artpi** [comment]
@meremagee, added free plan + domain scenario in this behaviour

**Comment by alisterscott** [comment]
@artpi: I noticed this behaviour doesn't happen for new users signing up for a plan via wordpress.com/create which uses the https://wordpress.com/start/premium flow. Would you like me to raise a bug for this?

**Pull request merged**